synapse/cmds/cortex.py

```diff
@@ -186,6 +186,65 @@ def openLogFd(self, opts):
         if off:
             return self.closeLogFd()
 
+class PsCmd(s_cli.Cmd):
+
+    _cmd_name = 'ps'
+    _cmd_syntax = ()
+
+    async def runCmdOpts(self, opts):
+
+        core = self.getCmdItem()
+        tasks = await core.ps()
+
+        for task in tasks:
+
+            self.printf('task iden: %s' % (task.get('iden'),))
+            self.printf('    name: %r' % (task.get('name'),))
+            self.printf('    user: %r' % (task.get('user'),))
+            self.printf('    status: %r' % (task.get('status'),))
+            self.printf('    metadata: %r' % (task.get('info'),))
+
+class KillCmd(s_cli.Cmd):
+    '''
+    Kill a running task/query within the cortex.
+
+    Syntax:
+        kill <iden>
+
+    Users may specify a partial iden GUID in order to kill
+    exactly one matching process based on the partial guid.
+    '''
+    _cmd_name = 'kill'
+    _cmd_syntax = (
+        ('iden', {}),
+    )
+
+    async def runCmdOpts(self, opts):
+
+        core = self.getCmdItem()
+
+        match = opts.get('iden')
+        if not match:
+            self.printf('no iden given to kill.')
+            return
+
+        idens = []
+        for task in await core.ps():
+            iden = task.get('iden')
+            if iden.startswith(match):
```

Comment:
issuing a single character here allows you to randomly kill 1 task whose guid starts with that character.  That's pretty dangerous.

Reply:
That's how git works...

Reply:
If that single character only matches one, i'm good with it.  That's the current paradigm for git / docker / etc.